kohese-portal/client/src/components/dashboard/project-dashboard/user-statistics/user-statistics.component.ts

```diff
@@ -273,12 +273,18 @@ export class UserStatisticsComponent extends NavigatableComponent
     }
   }
 
+  // uses hash lookups for primitives and linear search for objects
   findDuplicates(arr: Array<ItemProxy>) {
-    arr.filter((item,index) => {
-      if(arr.indexOf(item) != index) {
-        this.unassigned.splice(index,1);
-      }
+    let prims = {"boolean":{}, "number":{}, "string":{}}, objs = [];
+
+    arr.filter(function(item) {
+        let type = typeof item;
+        if(type in prims)
+            return prims[type].hasOwnProperty(item) ? false : (prims[type][item] = true);
```

Comment:
Is the false part of the ternary selector supposed to be an assignment?

Reply:
spoke about this on a call. The prims aren't being pushed to an array. The side-effect is believed to be inert. resolved?